testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/security/EjbSecurityDomainSetup.java

```diff
@@ -109,15 +113,22 @@ public void setup(final ManagementClient managementClient, final String containe
             applyUpdates(managementClient.getControllerClient(), Arrays.asList(compositeOp));
         } else {
             // elytron profile is enabled
-            ejbElytronDomainSetup = new EjbElytronDomainSetup(getUsersFile(), getGroupsFile(), getSecurityDomainName());
+            elytronDomainSetup = new ElytronDomainSetup(getUsersFile(), getGroupsFile(), getSecurityDomainName());
+            ejbElytronDomainSetup = new EjbElytronDomainSetup(getSecurityDomainName());
+            servletElytronDomainSetup = new ServletElytronDomainSetup(getSecurityDomainName());
+
+            elytronDomainSetup.setup(managementClient, containerId);
             ejbElytronDomainSetup.setup(managementClient, containerId);
+            servletElytronDomainSetup.setup(managementClient, containerId);
         }
     }
 
     @Override
     public void tearDown(final ManagementClient managementClient, final String containerId) {
-        if (ejbElytronDomainSetup != null) {
+        if (elytronDomainSetup != null) {
+            servletElytronDomainSetup.tearDown(managementClient, containerId);
```

Comment:
This multiple tearDown can affect another test cases. In case servletElytronDomainSetup.tearDown or ejbElytronDomainSetup.tearDown thrown exception then tearDown is not called on following ServerSetupTask (ejbElytronDomainSetup or elytronDomainSetup).

Reply:
I think once one of tearDowns fails, tearDown as whole will fail and following tests will be affected - nothing will improve if following tearDowns will be done.

Reply:
That's right in some cases but consider situation when ejbElytronDomainSetup.setup will fail for some reason (e.g. due to some bug) before it changes anything (i.e. configuration of server is not changed). Then if ejbElytronDomainSetup throws Exception in tearDown it causes that tearDown of elytronDomainSetup is not called (which means that configuration of server is not set to the original state). In this situation following test would not be affected if all tearDowns were called.

Reply:
On the other hand, most of tearDown in testsuit just ignore management operation failure - in such case just setting allowFailure=true in individual sub-tearDown would be appropriate...